pkg/cache/sql/informer/indexer.go

```diff
@@ -0,0 +1,255 @@
+type Indexer struct {
+	Store
+	indexers            cache.Indexers
+	indexersLock        sync.RWMutex
+	deleteIndicesStmt   *sql.Stmt
+	addIndexStmt        *sql.Stmt
+	listByIndexStmt     *sql.Stmt
+	listKeysByIndexStmt *sql.Stmt
+	listIndexValuesStmt *sql.Stmt
+}
+
+var _ cache.Indexer = (*Indexer)(nil)
+
+type Store interface {
+	DBClient
+	cache.Store
+
+	GetByKey(key string) (item any, exists bool, err error)
+	GetName() string
+	RegisterAfterUpsert(f func(key string, obj any, tx db.TXClient) error)
+	RegisterAfterDelete(f func(key string, tx db.TXClient) error)
+	GetShouldEncrypt() bool
+	GetType() reflect.Type
+}
+
+type DBClient interface {
+	Begin() (db.TXClient, error)
+	QueryForRows(ctx context.Context, stmt transaction.Stmt, params ...any) (*sql.Rows, error)
+	ReadObjects(rows db.Rows, typ reflect.Type, shouldDecrypt bool) ([]any, error)
+	ReadStrings(rows db.Rows) ([]string, error)
+	Prepare(stmt string) *sql.Stmt
+	CloseStmt(stmt db.Closable) error
+}
+
+// NewIndexer returns a cache.Indexer backed by SQLite for objects of the given example type
+func NewIndexer(indexers cache.Indexers, s Store) (*Indexer, error) {
+	// sanity checks first
+	for key := range indexers {
+		if strings.Contains(key, `"`) {
+			panic("Quote characters (\") in indexer names are not supported")
+		}
+	}
+
+	tx, err := s.Begin()
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Exec(fmt.Sprintf(createTableFmt, s.GetName()))
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Exec(fmt.Sprintf(createIndexFmt, s.GetName()))
+	if err != nil {
+		return nil, err
+	}
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	i := &Indexer{
+		Store:    s,
+		indexers: indexers,
+	}
+	i.RegisterAfterUpsert(i.AfterUpsert)
+
+	i.deleteIndicesStmt = s.Prepare(fmt.Sprintf(deleteIndicesFmt, s.GetName()))
+	i.addIndexStmt = s.Prepare(fmt.Sprintf(addIndexFmt, s.GetName()))
+	i.listByIndexStmt = s.Prepare(fmt.Sprintf(listByIndexFmt, s.GetName()))
+	i.listKeysByIndexStmt = s.Prepare(fmt.Sprintf(listKeyByIndexFmt, s.GetName()))
+	i.listIndexValuesStmt = s.Prepare(fmt.Sprintf(listIndexValuesFmt, s.GetName()))
+
+	return i, nil
+}
+
+/* Core methods */
+
+// AfterUpsert updates indices of an object
+func (i *Indexer) AfterUpsert(key string, obj any, tx db.TXClient) error {
+	// delete all
+	err := tx.StmtExec(tx.Stmt(i.deleteIndicesStmt), key)
+	if err != nil {
+		return err
+	}
+
+	// re-insert all
+	i.indexersLock.RLock()
+	defer i.indexersLock.RUnlock()
+	for indexName, indexFunc := range i.indexers {
+		values, err := indexFunc(obj)
+		if err != nil {
+			return err
+		}
+
+		for _, value := range values {
+			err = tx.StmtExec(tx.Stmt(i.addIndexStmt), indexName, value, key)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// add afterDelete
+/* Satisfy cache.Indexer */
+
+// Index returns a list of items that match the given object on the index function
+func (i *Indexer) Index(indexName string, obj any) ([]any, error) {
```

Comment:
Trying to use the Index method here: https://github.com/tomleb/lasso-controller-notes/blob/f4b0ad310f76999edfadaf03c82f5308f6cdf3b6/simple-vai/main.go#L87-L101 and it fails with 

```
[] sql: expected 1 destination arguments in Scan, not 4
```

I have tried using an upstream `cache.SharedIndexInformer` and the same example works fine. Am I missing something in the example that would make it work?